src/java.base/share/classes/java/text/MessageFormat.java

```diff
@@ -1651,6 +1779,88 @@ private static void copyAndFixQuotes(String source, int start, int end,
         }
     }
 
+    // Corresponding to the FormatType pattern
+    private enum FormatType {
+        NUMBER("number"),
+        DATE("date"),
+        TIME("time"),
+        J_DATE("j_date"),
+        J_TIME("j_time"),
+        CHOICE("choice"),
+        LIST("list"),
+
+        // Pre-defined DateTimeFormatter types
+        BASIC_ISO_DATE("basic_iso_date"),
+        ISO_LOCAL_DATE("iso_local_date"),
+        ISO_OFFSET_DATE ("iso_offset_date"),
+        ISO_DATE("iso_date"),
+        ISO_LOCAL_TIME("iso_local_time"),
+        ISO_OFFSET_TIME("iso_offset_time"),
+        ISO_TIME("iso_time"),
+        ISO_LOCAL_DATE_TIME("iso_local_date_time"),
+        ISO_OFFSET_DATE_TIME("iso_offset_date_time"),
+        ISO_ZONED_DATE_TIME("iso_zoned_date_time"),
+        ISO_DATE_TIME("iso_date_time"),
+        ISO_ORDINAL_DATE("iso_ordinal_date"),
+        ISO_WEEK_DATE("iso_week_date"),
+        ISO_INSTANT("iso_instant"),
+        RFC_1123_DATE_TIME("rfc_1123_date_time");
```

Comment:
Don't map these to lower case, keep the case exactly as declared in DateTimeFormatter.
It will be clearer that they are pre-defined constants.